modules/packages/Python.chpl

```diff
@@ -229,6 +228,94 @@
 */
 @unstable("The Python module's interface is under active development and may change")
 module Python {
+
+  /*
+  =======================
+  Developer Documentation
+  =======================
+
+  This module makes heavy usage of the Python C API. Care should be taken when
+  adding new calls to the C API, especially when it comes to reference counting
+  and the GIL.
+
+  Reference Counting
+  ------------------
+
+  Python objects are reference counted. When a Python object is created, it has
+  a reference count of 1. When the reference count reaches 0, the object is
+  deallocated. Normally, Python code that is interpreted has this handled by
+  the interpreter. Using the C API, requires the programmer to manage the
+  reference counts. This module hides this implementation detail from the user.
+
+  There are two things to keep in mind.
+    1. making sure that when Chapel is done with the object, the reference
+       count is zero.
+    2. making sure that while Chapel is still owning the object, the reference
+       count is greater than zero.
+
+  The `Value` base class has a field, `isOwned`, which determines if when the
+  Chapel object is deleted if the reference count will be decremented.
+  `toPython` creates new references and `fromPython` steals references
+  (decrements the reference count). Most other functions don't touch the
+  reference count. C API functions generally document if they return/take a new
+  reference ("strong reference"), a borrowed reference (reference count is
+  neither incremented or decremented), or a stolen reference (reference count is
+  decremented). Make sure to check the behavior of new functions.
+
+  Global Interpreter Lock (GIL)
+  -----------------------------
+
+  The Global Interpreter Lock (GIL) is a mutex that protects access to the
+  interpreter state. Some newer builds of Python are "free-threaded" and don't
+  have a GIL. Whether or not the GIL is present, this module will handle the
+  GIL. The `PyGILState_Ensure` and `PyGILState_Release` functions are used to
+  acquire and release the GIL, in free-threaded builds, these functions are
+  no-ops.
+
+  Python and non-python threads require special handling. The
+  `PyGILState_[Ensure|Release]` functions handle setting up new Python threads,
```

Comment:
Does `PyGILState_Release` set up a new Python thread?  I would also say that it should be called after calling Python code (we just write defer blocks so we don't fail to call it later)

Reply:
Yes, kinda. https://docs.python.org/3/c-api/init.html#non-python-created-threads. I reworked this whole paragraph, let me know if it reads better

Reply:
Looks good, thanks!